### ISSUE TITLE
Enable access to remote's LAN

### ISSUE THREAD

**Issue by thalst** [Jigsaw-Code/outline-client#939]
I want to be able to forward "local ip addresses" requests, such as 192.168.0.101 to the 
outline server, so that I'm able to access the remote LAN. This is useful because enables me to access some services running on my server without having to expose ports to the public. 

I've tried erasing everywhere on source where I found 192.168.x.x but it did not work.

Do you consider this a feature that could be implemented? I'm also happy if you point me in the direction of the source file I have to edit in able to achieve that. I'm using it on Linux with Electron.

**Comment by alalamav** [comment]
Hi @thalst, we err on the side of safety by protecting server's private addresses. This is why even though you have enabled your client to proxy the 192.168.x.x block, the server is not allowing the requests to go through.

If you understand the security implications, you can rebuild your Outline server by modifying the [target IP validator](https://github.com/Jigsaw-Code/outline-ss-server/blob/master/service/tcp.go#L131) in outline-ss-server.

**Comment by thalst** [comment]
Hi! Thanks for the answer. I'm actually using shadowsocks itself as the server. Should I modify it?

[Thiago Santos](https://spikenow.com/r/a/?ref=spike-organic-signature&_ts=v8cfu)	[v8cfu]
Graduando em Engenharia Elétrica---------------------------------------------------------------

Tel: +55 (31) 9 8962-9302	Universidade Federal de São João del-Rei
[youtube icon](http://lattes.cnpq.br/7833066199438085)[twitter icon](https://www.researchgate.net/profile/Thiago_Santos)
[tsantos.com.br](http://tsantos.com.br/)

On January 20, 2021 at 19:42 GMT, alalamav <notifications@github.com> wrote:

Hi [@thalst](https://github.com/thalst), we err on the side of safety by protecting server's private addresses. This is why even though you have enabled your client to proxy the 192.168.x.x block, the server is not allowing the requests to go through.

If you understand the security implications, you can rebuild your Outline server by modifying the [target IP validator](https://github.com/Jigsaw-Code/outline-ss-server/blob/master/service/tcp.go#L131) in outline-ss-server.

**Comment by alalamav** [comment]
> I'm actually using shadowsocks itself as the server. Should I modify it?

Do you mean shadowsocks-libev? If I remember correctly, it allows inbound LAN proxying.
On the client side, make sure you recompile and re-install the linux routing service. Otherwise your change to enable proxying the 192.168.x.x block won't take effect.

**Comment by thalst** [comment]
Exactly. Shadowsocks-libev allows that. I've made it work with their client. Anyway, could you give me some instructions about the file which I have to modify before compiling, and how to re-install linux routing service?

**Comment by cjhenck** [comment]
I believe this is the file? If this doesn't work, Alberto can confirm:
https://github.com/Jigsaw-Code/outline-client/blob/38cc535c8ffe3ee4580d9bc776d7d5c3fd0bf785/third_party/shadowsocks-libev/acl/server_block_local.acl

**Comment by flosoft** [comment]
Have you ever figured out how to access LAN with the Outline client?

**Comment by apollotsantos** [comment]
Hello,

Try this. In Windows proxy configuration, set the bypass addresses to fit
your needs. I've managed to do that with the shadowsocks client, but should
be the same.

Best,

Em sex, 13 de ago de 2021 08:00, Florian Jensen ***@***.***>
escreveu:

**Comment by chibikru** [comment]
I created 2 VPNs (two diffrent VPS servers). First I launched Outline (client). Then I launched OpenVPN (client). Got a local network - but without internet access))) but the local network works)))))))))

**Comment by siw1973** [comment]
@chibikru I did the same and got Local Network connectivity, but I would like to reroute Internet traffic through Outline -> OpenVPN -> Internet, have you tried this, come up with a solution ? Adding in "redirect-gateway def1" in my ovpn file doesn't seem to work as it would with just a OpenVPN -> Internet.
